client/client.go

```diff
@@ -143,3 +143,47 @@ func shouldTryLastPwd(lastPwd string, lastBindPasswordRotation time.Time) bool {
 	}
 	return lastBindPasswordRotation.Add(10 * time.Minute).After(time.Now())
 }
+
+func (c *Client) Add(cfg *Config, req *ldap.AddRequest) error {
+	if req == nil {
+		return fmt.Errorf("invalid request: request is nil")
+	}
+	if req.DN == "" {
+		return fmt.Errorf("invalid request: DN is empty")
+	}
+	conn, err := c.ldap.DialLDAP(cfg.ConfigEntry)
+	if err != nil {
+		return err
```

Comment:
Might want to wrap these e.g. `fmt.Errorf("error adding user: %w", err)` for better log context awareness, since the pattern is so similiar to `Del()`

Reply:
The context is usually added by the caller rather than inside the function otherwise it's easy to end up with `error adding user: error adding user: unable to dial LDAP server`

Reply:
👍